### DISCUSSION TITLE
Call API in Macros

### DISCUSSION THREAD

**Discussion by mattiamarras** [chatwoot/chatwoot#9855] [Self hosted] [answered]
It would be usefull make call to apis in macro workflows.
Now we make call in a Dashboard App, for example in order to rename the contact with the data of our CRM.

M.

**Comment by devlikepro** [comment, marked as answer]
For anyone looking for an alternative for API calls in Macros - you can run **n8n**, create **a workflow** there, and use its **Webhook URL** as a **Webhook action** in **Chatwoot** **Macros**. 

This way, you’ll have access to the full Chatwoot API (and other APIs like your CRM) directly from the workflow.

**Reply by sojan-official** [reply]
the recommended workflow here is to use webhook actions.
